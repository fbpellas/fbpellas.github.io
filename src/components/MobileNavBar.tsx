import * as React from 'react';
import { Nav, Navbar, NavDropdown } from 'react-bootstrap';
import { AUTHOR_FULLNAME } from '../constants';
import { MobileDropdownItem } from './MobileDropdownItem';

export interface MobileNavBarProps {
  setPageAndClear: (hash: string) => void;
}

const MobileNavBar: React.FC<MobileNavBarProps> = (props) => {
  const { setPageAndClear } = props;

  return (
    <Navbar sticky="top" collapseOnSelect expand="md" bg="dark" variant="dark">
      <Navbar.Brand href="#home">{AUTHOR_FULLNAME}</Navbar.Brand>
      <Navbar.Toggle aria-controls="responsive-navbar-nav" />
      <Navbar.Collapse id="responsive-navbar-nav">
        <Nav className="mr-auto">
          <NavDropdown title="About" id="collapsible-nav-dropdown-about">
            <MobileDropdownItem anchor="about-author" setPageAndClear={setPageAndClear} title="About the Author" />
            <MobileDropdownItem anchor="mission" setPageAndClear={setPageAndClear} title="Mission" />
          </NavDropdown>
          <NavDropdown title="Phonemes" id="collapsible-nav-dropdown-phonemes">
            <MobileDropdownItem anchor="phonemes" setPageAndClear={setPageAndClear} title="Phonemes" />
            <NavDropdown.Divider />
            <MobileDropdownItem anchor="vowels" setPageAndClear={setPageAndClear} title="Vowels" />
            <MobileDropdownItem anchor="diphthongs" setPageAndClear={setPageAndClear} title="Diphthongs" />
            <MobileDropdownItem anchor="consonants" setPageAndClear={setPageAndClear} title="Consonants" />
          </NavDropdown>
          <Nav.Link
            href="#stress"
            onClick={() => {
              setPageAndClear('stress');
            }}
          >
            Stress
          </Nav.Link>
          <NavDropdown title="Intonation" id="collapsible-nav-dropdown-intonation">
            <MobileDropdownItem anchor="intonation" setPageAndClear={setPageAndClear} title="Intonation" />
            <NavDropdown.Divider />
            <MobileDropdownItem anchor="falling" setPageAndClear={setPageAndClear} title="Falling" />
            <MobileDropdownItem anchor="rising" setPageAndClear={setPageAndClear} title="Rising" />
            <MobileDropdownItem anchor="non-final" setPageAndClear={setPageAndClear} title="Non-Final" />
          </NavDropdown>
          <Nav.Link
            href="#quiz"
            onClick={() => {
              setPageAndClear('quiz');
            }}
          >
            Quiz
          </Nav.Link>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

export { MobileNavBar };
