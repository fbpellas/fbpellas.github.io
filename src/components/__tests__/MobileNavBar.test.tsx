import React from 'react';
import { render } from '@testing-library/react';
import { MobileNavBar } from '../MobileNavBar';

test('MobileNavBar', () => {
  const fn = jest.fn();
  const { container } = render(<MobileNavBar setPageAndClear={fn} />);
  expect(container).toMatchSnapshot();
});
