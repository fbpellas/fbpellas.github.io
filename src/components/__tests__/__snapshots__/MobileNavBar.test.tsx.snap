// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavBar 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-dark bg-dark sticky-top"
  >
    <a
      class="navbar-brand"
      href="#home"
    >
      Faith Pellas
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        class="mr-auto navbar-nav"
      >
        <div
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            id="collapsible-nav-dropdown-about"
            role="button"
          >
            About
          </a>
        </div>
        <div
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            id="collapsible-nav-dropdown-phonemes"
            role="button"
          >
            Phonemes
          </a>
        </div>
        <a
          class="nav-link"
          data-rb-event-key="#stress"
          href="#stress"
        >
          Stress
        </a>
        <div
          class="dropdown nav-item"
        >
          <a
            aria-expanded="false"
            aria-haspopup="true"
            class="dropdown-toggle nav-link"
            href="#"
            id="collapsible-nav-dropdown-intonation"
            role="button"
          >
            Intonation
          </a>
        </div>
        <a
          class="nav-link"
          data-rb-event-key="#quiz"
          href="#quiz"
        >
          Quiz
        </a>
      </div>
    </div>
  </nav>
</div>
`;
